voluptuous/schema_builder.py

```diff
@@ -1036,6 +1038,7 @@ def __init__(
         self._schema = Schema(schema_)
         self.msg = msg
         self.description = description
+        self.__hash__ = lru_cache(maxsize=None)(lambda: hash(schema_))
```

Comment:
It seems like it would be lower overhead to use `functools.cache` here?

Reply:
Yes it would but it was only added in 3.9 and the tests are still running on 3.8

Reply:
Happy to do a PR to remove 3.8 support since it goes EOL in ~4 months anyways

Reply:
Let's do it, I'm okay with that.

Reply:
Actually I'm not sure it matters much since `cache` does the same thing under the hood and the native 
https://github.com/python/cpython/blob/1e7903d549716eb8034167302d4a718e42460b4b/Lib/functools.py#L653

It all end up back in native code either way https://github.com/python/cpython/blob/1e7903d549716eb8034167302d4a718e42460b4b/Lib/functools.py#L642C28-L642C46

Reply:
Still cleaner to use `@cache` though

Reply:
65ec763d67a18e3c43d7294f4ce6d902105da984